lib/stm32/common/dma_common_csel.c

```diff
@@ -0,0 +1,46 @@
+/** @addtogroup dma_file DMA peripheral API
+@ingroup peripheral_apis
+
+LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2019 Guillaume Revaillot <g.revaillot@gmail.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**@{*/
+
+#include <libopencm3/stm32/dma.h>
+
+/*---------------------------------------------------------------------------*/
+/** @brief DMA Channel Set Request Selection
+
+Set DMA request mapping selection for given channel. Refer to datasheet for channel
+request mapping tables.
+
+@param[in] dma unsigned int32. DMA controller base address: DMA1 or DMA2
+@param[in] channel unsigned int8. Channel number:  1-7 for DMA1 or 1-5 for DMA2
+@param[in] request unsigned int8. DMA request mapping.
```

Comment:
My only comment here, because you're doing such good stuff in general, is that the type information here in the doxygen is redundant.  the types are in the signature, and it just takes up space here in the comments that can be better used for the description.